Support static translations

aaronxsu opened the pull request:
## Overview

This PR
- adds `react-i18next` and relevant dependencies to support static translations
- updates app routing to include the language in the app URL
- refactors the language selection dropdown and language radio button group components to hook up with translation state
- updates the menu component with basic support of English, Spanish, and Chinese


### Checklist

- [ ] `fixup!` commits have been squashed
- [ ] `CHANGELOG.md` updated with summary of features or fixes, following
      [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines
- [ ] `README.md` updated if necessary to reflect the changes
- [ ] Run `./scripts/format` to lint, format, and fix the application source code.
- [ ] CI passes after rebase

### Demo

<img width="1216" height="971" alt="Screenshot 2025-07-12 at 6 44 20 PM" src="https://github.com/user-attachments/assets/cd37b6d6-2e5f-44f6-80a7-6e303f682ac1" />

<img width="1216" height="971" alt="Screenshot 2025-07-12 at 6 44 24 PM" src="https://github.com/user-attachments/assets/186d9acb-81b1-482b-a595-e3d4ccc20534" />

## Testing Instructions

 * Spin up the app and go to http://localhost:9966. Make sure the app URL adds the `/en/discover` segment automatically and the app is in English
 * Change the language with the top right dropdown. Make sure languages change in the top menu and the app URL changes too
 * Replace the last segment of the URL with `components`. On the components page, make sure the actively selected language in the URL, in the menu dropdown, and in the Radio Button Group are consistent. 
 * Use the language selection radio button or the dropdown. Make sure they change the language consistently


Resolves #640

aaronxsu committed:
Add frontend dep and configs for static translations

aaronxsu committed:
Update frontend router with language param

aaronxsu committed:
Refactor Menu component to integrate with language change

aaronxsu committed:
Refactor SelectLanguageButtons component to integrate with translation

rachelekm reviewed (changes requested):
Nice! The translations are working and updating on language code change and the URL is updating as well. There are two edge cases I found:
- I tried `http://localhost:9966/components` which took me to `http://localhost:9966/en/discover`. I would've expected it would navigate to `http://localhost:9966/en/components`, is that expected?

- I also notice if I try `http://localhost:9966/en/` it doesn't navigate to `http://localhost:9966/en/discover` as part of the catch-all

aaronxsu committed:
Improve app routing logic given languages

aaronxsu commented:
Those are great catches! I updated the handling of app routes. Please test and see if they cover those cases. I chose to do it in a more seemingly verbose but actually declarative way instead of handling everything in a `useEffect`, as the latter could become hard to debug very quickly.

Edit: this PR is now rebased on top of develop.

aaronxsu commented:
This is rebased on top of develop branch again.

aaronxsu committed:
Update changelog

rachelekm reviewed (approved):
Working great! Thanks for handling those edge cases this is good to go

aaronxsu commented:
Thanks for the review. Merging now.
